################################################################################
SHORT DESCRIPTION: 
################################################################################
XSA-169 (http://xenbits.xen.org/xsa/advisory-169.html)
Make debug output consistent in hvm_set_callback_via.

################################################################################
LONG DESCRIPTION: 
################################################################################
Source: http://xenbits.xen.org/xsa/advisory-169.html
Patches: xsa-169.patch

The unconditional printks in the switch statement of the hvm_set_callback_via
function results in Xen log spam in non debug versions of Xen. The printks
are for debug output only so conditionally compile the entire switch
statement on debug versions of Xen only.

################################################################################
PATCHES 
################################################################################
Index: xen-4.3.4/xen/arch/x86/hvm/irq.c
===================================================================
--- xen-4.3.4.orig/xen/arch/x86/hvm/irq.c	2015-03-19 16:08:36.000000000 +0100
+++ xen-4.3.4/xen/arch/x86/hvm/irq.c	2016-05-10 11:26:42.324939668 +0200
@@ -370,7 +370,8 @@
 
     spin_unlock(&d->arch.hvm_domain.irq_lock);
 
-    dprintk(XENLOG_G_INFO, "Dom%u callback via changed to ", d->domain_id);
+#ifndef NDEBUG
+    printk(XENLOG_G_INFO "Dom%u callback via changed to ", d->domain_id);
     switch ( via_type )
     {
     case HVMIRQ_callback_gsi:
@@ -386,6 +387,7 @@
         printk("None\n");
         break;
     }
+#endif
 }
 
 struct hvm_intack hvm_vcpu_has_pending_irq(struct vcpu *v)
